lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -30,6 +30,210 @@ using mlir::RegionRange;
 using namespace circt;
 using namespace firrtl;
 
+//===----------------------------------------------------------------------===//
+// Upstream to MLIR
+//===----------------------------------------------------------------------===//
+
+// The following should be part of `mlir::FunctionType`:
+
+/// Returns a new function type with the specified arguments and results
+/// inserted.
+FunctionType getWithArgsAndResults(FunctionType self,
```

Comment:
Agreed, this should be a method on FunctionType

Reply:
Upstreaming as https://reviews.llvm.org/D105011.